src/Stratis.Bitcoin.Features.Consensus/Rules/CommonRules/WitnessCommitmentsRule.cs

```diff
@@ -21,6 +21,14 @@ public class WitnessCommitmentsRule : PartialValidationConsensusRule
         /// <exception cref="ConsensusErrors.UnexpectedWitness">The block does not expect witness transactions but contains a witness transaction.</exception>
         public override Task RunAsync(RuleContext context)
         {
+            return this.ValidateWitnessCommitment(context, this.Parent.Network);
+        }
+
+        /// <summary>
+        /// Validation of the witness commitment if its found.
+        /// </summary>
+        public Task ValidateWitnessCommitment(RuleContext context, Network network)
```

Comment:
Just making a note here, we need an equivalent of the `CanCheckBlockWithWitness` PoW test for PoS. Especially so that we test both possible locations of the commitment.

Reply:
ah yes that's a good idea.

Reply:
done